internal/message/subtrie.go

```diff
@@ -118,7 +126,13 @@ func (t *Trie) Unsubscribe(ssid Ssid, subscriber Subscriber) {
 func (t *Trie) Lookup(ssid Ssid, filter func(s Subscriber) bool) (subs Subscribers) {
 	subs = newSubscribers()
 	t.RLock()
-	t.lookup(ssid, &subs, t.root, filter)
+
+	if "mqtt" == t.topicMatchingType {
+		t.mqttLookup(ssid, &subs, t.root, filter)
+	} else {
+		t.lookup(ssid, &subs, t.root, filter)
+	}
+
```

Comment:
We should not have a string matching in the lookup function, see the above comments. We can then simpy call `t.lookup ` here as usual.